### PR TITLE
Add support for trunc_32f to GLSL codegen. Updated the special_funcs tes...

### PR DESCRIPTION
**Pull request by abestephensg**
Add support for trunc_32f to GLSL codegen by adding a helper function that implements the operation plus testing in opengl/special_funcs. The test is refactored to provide more detailed error messages and wider bounds on undefined behavior of trig functions and type conversion.

### CODE REVIEW

**Comment by steven-johnson** [comment]
LGTM

File: test/opengl/special_funcs.cpp [hunk 1]
```diff
@@ -88,35 +90,65 @@ bool test_approx(Expr r, Expr g, Expr b, double rms_error) {
...
+    if (!test_exact<float>(trunc(x + 0.25f), trunc(-(x + 0.75f)), 0.0f)) {
+        printf("Failed trunc test\n");
+        errors++;
+    }
 
     // Trigonometric functions in GLSL are fast but not very accurate,
     // especially outside of 0..2pi.
     Expr r = (256 * x + y) / ceilf(65536.f / (2 * 3.1415926536f));
-    ok = test_approx<float>(sin(r), cos(r), 0, 5e-7);
+    if (!test_approx<float>(sin(r), cos(r), 0, 5e-2)) {
```
Comment (aam) [inline test/opengl/special_funcs.cpp]: Curious what is the reason for reduced accuracy?

**Comment by abestephensg** [comment]
The GLSL ES 1.0 spec does not define the precision of these functions and the error bound in the test worked for some implementations but not others. 

This was reported by bug: #594

**Comment by steven-johnson** [comment]
Can we add that as a comment?

**Comment by abestephensg** [comment]
Sure, I can elaborate the existing comment.

**Pull request merged**